app/models/resource_action_workflow.rb

```diff
@@ -95,6 +95,9 @@ def load_dialog(resource_action, values, options)
       dialog.target_resource = @target
       if options[:display_view_only]
         dialog.init_fields_with_values_for_request(values)
+      elsif options[:provision_workflow]
+        dialog.initialize_value_context(values)
+        dialog.load_values_into_fields(values)
```

Comment:
Testing this in a local setup I see the dialog field is initialized with `initialize_value_context` but the value gets reset to `nil` on `load_values_into_fields` when the caller does not pass in a value.

```ruby
user = User.current_user = User.find_by(:userid => "admin")
st = ServiceTemplate.find_by(:name => "Test Playbook Service")

# Good
vars = {:credential => nil, :hosts => "88.88.88.88"}
st.provision_request(User.current_user, vars)

# =>  options: {:dialog=>{"dialog_credential"=>nil, "dialog_hosts"=>"88.88.88.88", ...

# Bad
vars = {:credential => nil}
st.provision_request(User.current_user, vars)

# => options: {:dialog=>{"dialog_credential"=>nil, "dialog_hosts"=>nil, ...
```